### ISSUE TITLE
delivery_tag type mismatch between BlockingChannel and Channel

### ISSUE THREAD

**Issue by AndyMender** [pika/pika#1262]
The `BlockingChannel` class defines the following method:
```python
    def basic_reject(self, delivery_tag=None, requeue=True):
        """Reject an incoming message. This method allows a client to reject a
        message. It can be used to interrupt and cancel large incoming messages,
        or return untreatable messages to their original queue.

        :param int delivery-tag: The server-assigned delivery tag
        :param bool requeue: If requeue is true, the server will attempt to
                             requeue the message. If requeue is false or the
                             requeue attempt fails the messages are discarded or
                             dead-lettered.

        """
        self._impl.basic_reject(delivery_tag=delivery_tag, requeue=requeue)
        self._flush_output()
```
Which in turn calls the corresponding method of the `Channel` class:
```python

    def basic_reject(self, delivery_tag, requeue=True):
        """Reject an incoming message. This method allows a client to reject a
        message. It can be used to interrupt and cancel large incoming messages,
        or return untreatable messages to their original queue.

        :param integer delivery-tag: int/long The server-assigned delivery tag
        :param bool requeue: If requeue is true, the server will attempt to
                             requeue the message. If requeue is false or the
                             requeue attempt fails the messages are discarded or
                             dead-lettered.
        :raises: TypeError

        """
        self._raise_if_not_open()
        if not is_integer(delivery_tag):
            raise TypeError('delivery_tag must be an integer')
        return self._send_method(spec.Basic.Reject(delivery_tag, requeue))
```
A call to `BlockingChannel.basic_reject` will throw a `TypeError` exception if the `delivery_tag` is not defined (`None` by default) or has a non-integer value (for example, an instance of `pika.spec.Basic.Deliver`). This problem concerns multiple other methods in which `delivery_tag` does not default to an integer value or is not a positional argument.

**Comment by lukebakken** [comment]
Seems like an oversight. We would happily accept a pull request to fix this.

**Comment by AndyMender** [comment]
@lukebakken Apologies. I should've opened a pull request as soon as I had found the bug. Here it is: https://github.com/pika/pika/pull/1264

**Comment by AndyMender** [comment]
I completely forgot about this issue, but can the associated PR be merged? @lukebakken ?

**Comment by lukebakken** [comment]
Fixed by #1264. Thanks!
